client/components/widget/query/builder/query-builder-datasource-config-directive_test.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * @copyright Copyright 2014 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @fileoverview Tests for QueryBuilderDatasourceConfigDirective, which encapsulates the global config
+ * settings.
+ * @author joemu@google.com (Joe Allan Muharsky)
+ */
+
+goog.require('p3rf.perfkit.explorer.components.config.ConfigService');
+goog.require('p3rf.perfkit.explorer.components.dashboard.DashboardService');
+goog.require('p3rf.perfkit.explorer.components.widget.query.builder.QueryBuilderDatasourceConfigDirective');
+goog.require('p3rf.perfkit.explorer.models.ChartWidgetModel');
+
+describe('QueryBuilderDatasourceConfigDirective', function() {
```

Comment:
It's really been following convention, but only because it predated the standard/suggestion.  Will make a future PR to clean this up.